File: examples/dart/language_gauntlet/lib/src/runtime/executor.dart
import 'dart:io';

String execute(String cmd) {
  // SINK — Process.runSync with runInShell: true · dart.cmdi.process_runsync_in_shell · CWE-78
  Process.runSync('sh', ['-c', cmd], runInShell: true);
  return cmd;
}

String cleanTwin() {
  // NEGATIVE — same sink kind with a constant argument must not report.
  Process.runSync('sh', ['-c', 'echo clean'], runInShell: true);
  return 'clean';
}

File: examples/rust/language_gauntlet/src/runtime/executor.rs
use std::process::Command;

pub fn execute(cmd: &str) -> String {
    // SINK — rust.cmdi.command_new with tainted cmd. Broken into
    // separate statements so the adapter reliably attributes the
    // Command::new call to this function rather than module scope.
    let mut c = Command::new("sh");
    c.arg("-c");
    c.arg(cmd);
    let _ = c.status();
    cmd.to_string()
}

pub fn clean_twin() -> String {
    // NEGATIVE — same sink kind with a constant argument must not report.
    let mut c = Command::new("sh");
    c.arg("-c");
    c.arg("echo clean");
    let _ = c.status();
    "clean".to_string()
}
